File: app/src/main/java/com/ipd/taxiu/adapter/OrderListAdapter.java
package com.ipd.taxiu.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.ipd.taxiu.R;
import com.ipd.taxiu.bean.OrderBean;
import com.ipd.taxiu.ui.activity.order.LogisticsDetailActivity;
import com.ipd.taxiu.ui.activity.order.OrderDetailActivity;

import java.util.ArrayList;

/**
 * Created by Miss on 2018/7/19
 */
public class OrderListAdapter extends RecyclerView.Adapter<OrderListAdapter.ViewHolder> {
    public static String PAYMENT = "付款";
    public static String DETAIL = "详情";
    public static String DELIVERY = "收货";
    public static String EVALUATE = "评价";

    private Context context;
    private ArrayList<OrderBean> list;

    public OrderListAdapter(Context context, ArrayList<OrderBean> list) {
        this.context = context;
        this.list = list;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_order_list, parent, false);
        return new ViewHolder(view);
    }

    @SuppressLint("ResourceAsColor")
    @Override
    public void onBindViewHolder(final OrderListAdapter.ViewHolder holder, int position) {
        switch (list.get(position).getStatus()) {
            case 1:
                holder.order_status.setText("待付款");
                holder.confirm.setText("付款");
                holder.cancel.setText("取消");
                holder.confirm.setVisibility(View.VISIBLE);
                holder.cancel.setVisibility(View.VISIBLE);
                break;
            case 2:
                holder.order_status.setText("待发货");
                holder.cancel.setVisibility(View.GONE);
                holder.confirm.setText("详情");
                holder.confirm.setVisibility(View.VISIBLE);
                holder.confirm.setBackgroundResource(R.drawable.shape_order_btn_cancel);
                holder.confirm.setTextColor(context.getResources().getColor(R.color.black));
                break;
            case 3:
                holder.order_status.setText("待收货");
                holder.confirm.setText("收货");
                holder.cancel.setText("物流");
                holder.confirm.setVisibility(View.VISIBLE);
                holder.cancel.setVisibility(View.VISIBLE);
                break;
            case 4:
                holder.order_status.setText("已完成");
                holder.confirm.setText("评价");
                holder.cancel.setText("再次购买");
                holder.confirm.setVisibility(View.VISIBLE);
                holder.cancel.setVisibility(View.VISIBLE);
                break;
            default:
                break;
        }

        final String confirmText = holder.confirm.getText().toString();
        holder.confirm.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent;
                if (confirmText.equals("付款")) {
                    intent = new Intent(context, OrderDetailActivity.class);
                    intent.putExtra("order_status", PAYMENT);
                    context.startActivity(intent);
                } else if (confirmText.equals("详情")) {
                    intent = new Intent(context, OrderDetailActivity.class);
                    intent.putExtra("order_status", DETAIL);
                    context.startActivity(intent);
                } else if (confirmText.equals("收货")) {
                    intent = new Intent(context, OrderDetailActivity.class);
                    intent.putExtra("order_status", DELIVERY);
                    context.startActivity(intent);
                } else if (confirmText.equals("评价")) {
                    intent = new Intent(context, OrderDetailActivity.class);
                    intent.putExtra("order_status", EVALUATE);
                    context.startActivity(intent);
                }
            }
        });

        final String cancelText = holder.cancel.getText().toString();
        holder.cancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (cancelText.equals("取消")) {
                } else if (cancelText.equals("物流")) {
                    Intent intent = new Intent(context,LogisticsDetailActivity.class);
                    context.startActivity(intent);
                } else if (cancelText.equals("再次购买")) {
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        TextView order_status, confirm, cancel;

        public ViewHolder(View itemView) {
            super(itemView);
            order_status = itemView.findViewById(R.id.commodity_status);
            confirm = itemView.findViewById(R.id.tv_confirm);
            cancel = itemView.findViewById(R.id.tv_cancel);
        }

    }

}

File: app/src/main/java/com/ipd/taxiu/adapter/OrderListAdapter.kt
package com.ipd.taxiu.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.ipd.taxiu.R
import com.ipd.taxiu.bean.OrderBean
import com.ipd.taxiu.utils.Order
import kotlinx.android.synthetic.main.item_order_list.view.*

/**
 * Created by Miss on 2018/7/19
 */
class OrderListAdapter(private val context: Context, private val list: List<OrderBean>?, private val itemClickListener: Order.OrderItemClickListener) : RecyclerView.Adapter<OrderListAdapter.ViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_order_list, parent, false))
    }

    override fun onBindViewHolder(holder: OrderListAdapter.ViewHolder, position: Int) {
        val info = list!![position]
        holder.itemView.tv_order_number.text = info.ORDER_NO

        if (info.status == Order.PAYMENT) {
            holder.itemView.commodity_pay.text = "应付(含运费) :"
            holder.itemView.tv_commodity_pay.text = "￥" + info.PAYABLE_FEE
        } else {
            holder.itemView.commodity_pay.text = "实付(含运费) :"
            holder.itemView.tv_commodity_pay.text = "￥" + info.PAY_FEE
        }

        holder.itemView.product_recycler_view.adapter = OrderProductAdapter(context, info.PRODUCT_LIST, {
            itemClickListener.onItemClick(info)
        })
        holder.itemView.setOnClickListener {
            itemClickListener.onItemClick(info)
        }


        when (info.status) {
            Order.PAYMENT -> {
                holder.itemView.order_status.text = "待付款"
                holder.itemView.tv_confirm.text = "付款"
                holder.itemView.tv_cancel.text = "取消"
                holder.itemView.tv_confirm.visibility = View.VISIBLE
                holder.itemView.tv_cancel.visibility = View.VISIBLE

                holder.itemView.tv_confirm.setOnClickListener {
                    //付款
                    itemClickListener.onPayment(info)
                }
                holder.itemView.tv_cancel.setOnClickListener {
                    //取消
                    itemClickListener.onCancelOrder(info)
                }
            }
            Order.WAIT_SEND -> {
                holder.itemView.order_status.text = "待发货"
                holder.itemView.tv_cancel.visibility = View.GONE
                holder.itemView.tv_confirm.text = "详情"
                holder.itemView.tv_confirm.visibility = View.VISIBLE
//                holder.itemView.tv_confirm.setBackgroundResource(R.drawable.shape_order_btn_cancel)
//                holder.itemView.tv_confirm.setTextColor(context.resources.getColor(R.color.black))

                holder.itemView.tv_confirm.setOnClickListener {
                    //详情
                    itemClickListener.onItemClick(info)
                }
                holder.itemView.tv_cancel.setOnClickListener {
                    //null
                }
            }
            Order.WAIT_RECEIVE -> {
                holder.itemView.order_status.text = "待收货"
                holder.itemView.tv_confirm.text = "收货"
                holder.itemView.tv_cancel.text = "物流"
                holder.itemView.tv_confirm.visibility = View.VISIBLE
                holder.itemView.tv_cancel.visibility = View.VISIBLE

                holder.itemView.tv_confirm.setOnClickListener {
                    //收货
                    itemClickListener.onReceived(info)
                }
                holder.itemView.tv_cancel.setOnClickListener {
                    //物流
                    itemClickListener.onExpress(info)
                }
            }
            Order.EVALUATE -> {
                holder.itemView.order_status.text = "待评价"
                holder.itemView.tv_confirm.text = "评价"
                holder.itemView.tv_cancel.text = "再次购买"
                holder.itemView.tv_confirm.visibility = View.VISIBLE
                holder.itemView.tv_cancel.visibility = View.VISIBLE

                holder.itemView.tv_confirm.setOnClickListener {
                    //评价
                    itemClickListener.onEvaluate(info)
                }
                holder.itemView.tv_cancel.setOnClickListener {
                    //再次购买
                    itemClickListener.onBuyAgain(info)
                }
            }
            Order.FINFISH -> {
                holder.itemView.order_status.text = "已完成"
                holder.itemView.tv_confirm.text = "再次购买"
                holder.itemView.tv_confirm.visibility = View.VISIBLE
                holder.itemView.tv_cancel.visibility = View.GONE

                holder.itemView.tv_confirm.setOnClickListener {
                    //再次购买
                    itemClickListener.onBuyAgain(info)
                }
                holder.itemView.tv_cancel.setOnClickListener {
                }
            }
            else -> {
            }
        }

//        val tv_confirmText = holder.itemView.tv_confirm.text.toString()
//        holder.itemView.tv_confirm.setOnClickListener {
//            val intent: Intent
//            when (tv_confirmText) {
//                "付款" -> {
//                    intent = Intent(context, OrderDetailActivity::class.java)
//                    intent.putExtra("order_status", Order.PAYMENT)
//                    context.startActivity(intent)
//                }
//                "详情" -> {
//                    intent = Intent(context, OrderDetailActivity::class.java)
//                    intent.putExtra("order_status", Order.WAIT_SEND)
//                    context.startActivity(intent)
//                }
//                "收货" -> {
//                    intent = Intent(context, OrderDetailActivity::class.java)
//                    intent.putExtra("order_status", Order.WAIT_RECEIVE)
//                    context.startActivity(intent)
//                }
//                "评价" -> {
//                    intent = Intent(context, OrderDetailActivity::class.java)
//                    intent.putExtra("order_status", Order.EVALUATE)
//                    context.startActivity(intent)
//                }
//            }
//        }
//
//        val tv_cancelText = holder.itemView.tv_cancel.text.toString()
//        holder.itemView.tv_cancel.setOnClickListener {
//            when (tv_cancelText) {
//                "取消" -> initMessageDialog(context as Activity)
//                "物流" -> {
//                    val intent = Intent(context, LogisticsDetailActivity::class.java)
//                    context.startActivity(intent)
//                }
//                "再次购买" -> {
//                }
//            }
//        }
    }

    override fun getItemCount() = list?.size ?: 0

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)

}
